Code/Framework/AzCore/AzCore/Math/Sfmt.cpp

```diff
@@ -515,21 +517,16 @@ namespace AZ
     //=========================================================================
     AZ::u32 Sfmt::Rand32()
     {
-        int index = m_index.fetch_add(1);
-        if (index >= SfmtInternal::N32)
+        AZStd::lock_guard<decltype(m_generationMutex)> lock(m_generationMutex);
```

Comment:
Class Template Argument deduction can be used here.
```suggestion
        AZStd::scoped_lock lock(m_generationMutex);
```